File: duna-core/src/main/java/io/duna/core/concurrent/DunaThread.java
package io.duna.core.concurrent;

import io.netty.channel.EventLoop;
import org.jetbrains.annotations.NonNls;

public class DunaThread extends Thread {

    private EventLoop eventLoop;

    public DunaThread(Runnable runnable, @NonNls String s) {
        super(runnable, s);
    }

    public EventLoop getEventLoop() {
        return eventLoop;
    }

    void setEventLoop(EventLoop eventLoop) {
        this.eventLoop = eventLoop;
    }
}

File: duna-core/src/main/kotlin/io/duna/core/concurrent/execution/DunaThread.kt
package io.duna.core.concurrent.execution

import java.util.concurrent.ExecutorService

class DunaThread(block: Runnable,
                 name: String) : Thread(block, "duna-$name") {

  internal lateinit var executorService: ExecutorService

  companion object {
    @JvmStatic
    fun currentThread(): DunaThread? {
      if (Thread.currentThread() is DunaThread)
        return Thread.currentThread() as DunaThread

      return null
    }
  }
}
